docs: DD-523 Wallets > Wallet Workflows (Overview) (#3309)
dfinity/portal: docs/developer-docs/defi/wallets/workflow.mdx

---

import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";

# Wallet overview

<MarkdownChipRow labels={["Beginner" , "Concept"]} />

Users and developers on ICP use wallets for the following activities:

- Holding, transferring, and receiving assets such as fungible including ICP and other ICRC1/ICRC2 tokens (ex. ckBTC, ckETH) and non-fungible tokens.
- Deploying, upgrading, and performing operations on canisters.
- Authenticating with decentralized applications.

On other networks, these activities may all use the same wallet account and workflow. On ICP, each of these activities will likely use a separate wallet address or service. This document provides an overview of each different workflow, the type of account associated with it, and how to interact with it.

## Types of wallets 

On ICP, both developers and end users will use different types of wallets for different interactions. Here is a table that describes each type of wallet and the activities associated with them: 

 <table>
    <tr>
        <th>Type of wallet</th>
        <th>Activities</th>
        <th>Examples</th>
    </tr>
    <tr>

>         <td>[Asset wallet](/overview.mdx)</td>

Reviewer: ```suggestion
        <td>[Asset wallet](overview.mdx)</td>
```

---
keywords: [beginner, concept, wallet, wallets, authentication, assets]
---

import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";

> # Wallet Overview

Reviewer: ```suggestion
# Wallet overview
```

<MarkdownChipRow labels={["Beginner" , "Concept"]} />

> This document provides an overview of the purpose of different types of wallets on the Internet Computer and how to interact with them. 
> 
> Users of the Internet Computer use wallets for the following activities:
> 
> - Hold, transfer, and receive assets such as fungible including ICP and other ICRC1/ICRC2 tokens (ex. ckBTC, ckETH) and non-fungible tokens.
> - Deploy, upgrade, and perform operations on canisters.
> 
> Decentralized applications developed on ICP can authenticate users using wallets or [Internet Identity](/docs/current/developer-docs/identity/internet-identity/overview).
> 
> Though not directly considered a wallet, Internet Identity behind the scenes generates a principal. This principal not only identifies you to various user-facing dapps but is also an identifier to send and receive assets on ICP.

Reviewer: ```suggestion
Users and developers on ICP use wallets for the following activities:

- Holding, transferring, and receiving assets such as fungible including ICP and other ICRC1/ICRC2 tokens (ex. ckBTC, ckETH) and non-fungible tokens.
- Deploying, upgrading, and performing operations on canisters.
- Authenticating with decentralized applications.

On other networks, these activities may all use the same wallet account and workflow. On ICP, each of these activities will likely use a separate wallet address or service. This document provides an overview of each different workflow, the type of account associated with it, and how to interact with it.
```

Author: Thank you for this wording. This definitely makes it more clear to developers and users from other ecosystems.

> ## Types of Wallets 

Reviewer: ```suggestion
## Types of wallets 
```

> On ICP, users may use different types of wallets for different types of interactions. Here is a table that describes the types of wallets based on interaction:

Reviewer: ```suggestion
On ICP, both developers and end users will use different types of wallets for different interactions. Here is a table that describes each type of wallet and the activities associated with them: 
```

 <table>
    <tr>

>         <th>Type of Wallet</th>

Reviewer: ```suggestion
        <th>Type of wallet</th>
```

        <th>Activities</th>
        <th>Examples</th>
    </tr>
    <tr>

>         <td>[Asset Wallet](/overview)</td>

Reviewer: ```suggestion
        <td>[Asset wallet](/overview.mdx)</td>
```

>         <td> - Hold, transfer, and receive assets such as fungible tokens including ICP and other ICRC1/ICRC2 tokens (ex. ckBTC, ckETH) and non-fungible tokens 
>         <br>
>         - View token holdings
>         <br>
>         - Stake ICP and create a neuron to participate in the NNS
>         <br>
>         - Can also be used for user based authentication
>         </td>
>         <td> - Plug, Stoic, Oisy. Please refer to the [Asset Wallets docs for the full list](/docs/current/developer-docs/defi/wallets/overview#third-party-custody) </td>

Reviewer: ```suggestion
        <td> - Hold, transfer, and receive assets such as fungible tokens including ICP and other ICRC1/ICRC2 tokens (ex. ckBTC, ckETH) and non-fungible tokens.
        <br>
        - View token holdings.
        <br>
        - Stake ICP and create a neuron to participate in the NNS.
        <br>
        - User authentication.
        </td>
        <td> - Plug, Stoic, Oisy. Please refer to the [asset wallets docs for the full list](/docs/current/developer-docs/defi/wallets/overview#third-party-custody) </td>
```

Author: Commit 2969d025d

    </tr>
    <tr>

>         <td>[Canisters](/docs/current/developer-docs/smart-contracts/overview/introduction)</td>

Reviewer: ```suggestion
        <td>[Canister-controlled wallet](/docs/current/developer-docs/smart-contracts/overview/introduction)</td>
```

        <td>Using [Chain Fusion](/docs/current/developer-docs/multi-chain/overview), canisters on ICP can hold, sign, and transact with assets on various chains including ICP, Bitcoin, Ethereum, EVMs, and more.
        </td>
        <td>Any canister</td>
    </tr>
    <tr>

>         <td>[Cycles](//docs/current/developer-docs/defi/cycles/cycles-wallet) (will be depreciated from dfx) </td>

Reviewer: ```suggestion
        <td>[Cycles wallet](//docs/current/developer-docs/defi/cycles/cycles-wallet) (will be depreciated from dfx) </td>
```

>         <td> - Holds cycles needed to deploy, upgrade, and perform other operations on canisters
>         <br>
>         - The Cycles Wallet will be replaced by the Cycles ledger. By using the Cycles ledger, cycles will compile to common ICRC1/ICRC2/ICRC3 standards.
>         </td>
>         <td>The Cycles Wallet will be replaced by the [Cycles ledger](docs/current/developer-docs/defi/cycles/cycles-ledger). By using the Cycles ledger, cycles will compile to common ICRC1/ICRC2/ICRC3 standards.</td>

Reviewer: ```suggestion
        <td> - Holds cycles needed to deploy, upgrade, and perform other operations on canisters.
        <br>
        - The cycles wallet is still supported, but has been replaced by the [cycles ledger](docs/current/developer-docs/defi/cycles/cycles-ledger). It is recommended to use the cycles ledger instead.
        </td>
```

    </tr>
</table>

### Other authentication methods

> As mentioned, Internet Identity is not directly considered a wallet but more of an authentication method. However, an Internet Identity is tied to a unique principal for each unique domain that can be used to hold and receive assets. 

Reviewer: ```suggestion
Internet Identity is not directly considered a wallet but more of an authentication method. However, an Internet Identity is tied to a unique principal for each unique domain that can be used to hold and receive assets. 
```
